packages/@aws-cdk/aws-servicecatalog/lib/product-stack.ts

```diff
@@ -60,8 +72,38 @@ export class ProductStack extends cdk.Stack {
       fileName: this.templateFile,
     }).httpUrl;
 
+    if (this._productVersionDetails.locked !== undefined) {
+      this.writeTemplateToSnapshot(cfn);
+    }
```

Comment:
Why doesn't the `History` class write and validate the snapshots?

Reply:
I think it could, it would involve giving the productStack access to it's `History` class and calling a method from it, I'll try to implement it.